backend/core/src/main/java/org/sonarsource/sonarlint/core/analysis/UserAnalysisPropertiesRepository.java

```diff
@@ -40,11 +39,23 @@ public Map<String, String> getUserProperties(String configurationScopeId) {
     return properties;
   }
 
-  public void setUserProperties(String configurationScopeId, Map<String, String> extraProperties) {
-    propertiesByConfigScope.put(configurationScopeId, new HashMap<>(extraProperties));
+  public boolean setUserProperties(String configurationScopeId, Map<String, String> extraProperties) {
+    var oldProperties = propertiesByConfigScope.get(configurationScopeId);
+    var newProperties = new HashMap<>(extraProperties);
+    var changed = !newProperties.equals(oldProperties);
+    if (changed) {
+      propertiesByConfigScope.put(configurationScopeId, newProperties);
+    }
+    return changed;
```

Comment:
I am not sure there is a need to create a new HashMap. What about something a bit simpler: 

```
var oldProperties = propertiesByConfigScope.put(configurationScopeId, extraProperties);
return !extraProperties.equals(oldProperties);
```

This should also play nicer with concurrent access than get and put in 2 different steps.

Reply:
That's correct. But if anything, I would change `getUserProperties()` to create a new `HashMap`. It's a detail, you could also keep adding a new map to propertiesByConfigScope, but still simplify the body. Something like: 

```
var oldProperties = propertiesByConfigScope.put(configurationScopeId, new HashMap(extraProperties));
return !extraProperties.equals(oldProperties);
```